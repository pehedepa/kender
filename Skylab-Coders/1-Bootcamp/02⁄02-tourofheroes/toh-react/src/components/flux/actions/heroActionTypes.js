const heroActionTypes = {
  ADD_HERO: 'ADD_HERO',
  DELETE_HERO: 'DELETE_HERO',
  UPDATE_HERO: 'UPDATE HERO',
  LOAD_HEROES: 'LOAD HEROES',
};

export default heroActionTypes;
