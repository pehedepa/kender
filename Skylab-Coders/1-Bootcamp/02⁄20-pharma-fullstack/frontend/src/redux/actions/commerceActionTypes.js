export default {
  LOAD_ITEMS: 'LOAD_ITEMS',
  ADD_ITEM_TO_CART: 'ADD_ITEM_TO_CART'
};
