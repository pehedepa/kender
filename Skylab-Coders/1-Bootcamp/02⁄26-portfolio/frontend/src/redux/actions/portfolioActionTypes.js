export default {
  GET_GENERIC: 'GET_GENERIC',
  GET_ACADEMIC: 'GET_ACADEMIC',
  GET_TRADESKILLS: 'GET_TRADESKILLS',
  GET_WORKEXPERIENCE: 'GET_WORKEXPERIENCE'
};
