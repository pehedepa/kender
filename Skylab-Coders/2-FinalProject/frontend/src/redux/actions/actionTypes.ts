export default {
  ADD_USER: 'ADD_USER',
  ADD_USER_ERROR: 'ADD_USER_ERROR',
  CLEAN_REGISTER_STATUS: 'CLEAN_REGISTER_STATUS',
  CLEAN_LOGIN_STATUS: 'CLEAN_LOGIN_STATUS',
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_ERROR: 'LOGIN_ERROR',
  UPLOAD_PRODUCT: 'UPLOAD_PRODUCT',
  CLEAN_UPLOAD_PRODUCT: 'CLEAN_UPLOAD_PRODUCT',
  GET_ALL_PRODUCTS: 'GET_ALL_PRODUCTS',
  GET_ALL_PRODUCTS_ERROR: 'GET_ALL_PRODUCTS_ERROR',
  FILTER_PRODUCTS: 'FILTER_PRODUCTS',
  CLEAN_FILTERS: 'CLEAN_FILTERS',
  UPDATE_PROFILE: 'UPDATE_PROFILE',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  LOG_OUT: 'LOG_OUT'
}
