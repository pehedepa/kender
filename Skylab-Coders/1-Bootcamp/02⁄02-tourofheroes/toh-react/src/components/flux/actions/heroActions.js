import dispatcher from '../dispatcher/dispatcher';
import heroActionTypes from './heroActionTypes';

export function saveHero(hero) {
  debugger;
  const action = {
    type: hero.id ? heroActionTypes.UPDATE_HERO : heroActionTypes.ADD_HERO,
    hero,
  };

  dispatcher.dispatch(action);
}

export function deleteHero({ id }) {
  debugger;
  const action = {
    type: heroActionTypes.DELETE_HERO,
    heroId: id,
  };

  dispatcher.dispatch(action);
}

export function loadHero(heroId) {
  const action = {
    type: heroActionTypes.LOAD_HERO,
    heroId,
  };

  dispatcher.dispatch(action);
}
