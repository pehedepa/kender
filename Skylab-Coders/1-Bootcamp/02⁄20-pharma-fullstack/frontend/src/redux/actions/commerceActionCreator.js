import axios from 'axios';
import commerceActionTypes from './commerceActionTypes';

const api = 'http://localhost:5000/api/commerce';

export default function getItems() {
  return async (dispatch) => {
    const { data } = await axios(api);
    dispatch({
      type: commerceActionTypes.LOAD_ITEMS,
      data
    });
  };
}

export function pushItemToCart(name, _id, url, stock, price) {
  const payload = {
    name,
    _id,
    url,
    stock,
    price
  };

  return (dispatch) => {
    dispatch({
      type: commerceActionTypes.ADD_ITEM_TO_CART,
      data: payload
    });
  };
}
