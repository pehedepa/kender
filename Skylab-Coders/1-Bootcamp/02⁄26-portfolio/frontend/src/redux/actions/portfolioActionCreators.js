import axios from 'axios';
import portfolioActionTypes from './portfolioActionTypes';

const api = 'http://localhost:5000/api/portfolio';

export default function getGenericData() {
  const generic = '/generic';
  return async (dispatch) => {
    const { data } = await axios(api + generic);
    dispatch({
      type: portfolioActionTypes.GET_GENERIC,
      data
    });
  };
}

export function getTradeSkillsData() {
  const tradeSkills = '/tradeskills';
  return async (dispatch) => {
    const { data } = await axios(api + tradeSkills);
    dispatch({
      type: portfolioActionTypes.GET_TRADESKILLS,
      data
    });
  };
}
