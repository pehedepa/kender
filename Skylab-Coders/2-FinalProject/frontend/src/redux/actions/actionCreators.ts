import axios from 'axios'
import actionTypes from './actionTypes'
import { loginURL, registerURL, productsURL, usersURL, logoutURL } from '../../constants'
import { Dispatch } from 'react'

export function login (email: string, password: string): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(loginURL, data)
      dispatch({
        type: actionTypes.LOGIN_USER,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOGIN_ERROR,
        payload: 401
      })
    }
  }
}

export function createAcc (email: string, password: string, location: string, name: string): Promise<any> {
  const data = { email, password, location, name }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(registerURL, data)
      dispatch({
        type: actionTypes.ADD_USER,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.ADD_USER_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProduct (name: string, price: number, description: string, image: string, user: string, category: string, ageRange: string, keywords: any[]): Promise<any> {
  const data = { name, price, description, image, user, category, ageRange, keywords }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.post(productsURL, data)
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: response.status
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPLOAD_PRODUCT,
        payload: 401
      })
    }
  }
}

export function getAllProducts (): Promise<any> {
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.get(productsURL)
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS,
        payload: response.data
      })
    } catch (error) {
      dispatch({
        type: actionTypes.GET_ALL_PRODUCTS_ERROR,
        payload: 401
      })
    }
  }
}

export function uploadProfile (_id, photo, name, address, cc, location): Promise<any> {
  const data = { _id, photo, name, address, cc, location }
  return async function dispatchInfo (dispatch) {
    try {
      const response = await axios.put(usersURL, data)
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: response
      })
    } catch (error) {
      dispatch({
        type: actionTypes.UPDATE_PROFILE,
        payload: 401
      })
    }
  }
}

export function logout (email, password): Promise<any> {
  const data = { email, password }
  return async function dispatchInfo (dispatch) {
    try {
      await axios.post(logoutURL, data)
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    } catch (error) {
      dispatch({
        type: actionTypes.LOG_OUT,
        payload: null
      })
    }
  }
}

export function deleteProduct (_id) {
  return async function dispatchInfo (dispatch) {
    try {
      await axios.delete(`${productsURL}/${_id}`)
      dispatch({
        type: actionTypes.DELETE_PRODUCT
      })
    } catch (error) {
      dispatch({
        type: actionTypes.DELETE_PRODUCT,
        payload: null
      })
    }
  }
}

export function applyFilter (data: string) : { type: string, payload: string } {
  return {
    type: actionTypes.FILTER_PRODUCTS,
    payload: data
  }
}

export function cleanFilters (): { type: string, payload: null } {
  return {
    type: actionTypes.CLEAN_FILTERS,
    payload: null
  }
}

export function cleanRegister (): { type: string } {
  return {
    type: actionTypes.CLEAN_REGISTER_STATUS
  }
}

export function cleanLogin (): { type: string } {
  return {
    type: actionTypes.CLEAN_LOGIN_STATUS
  }
}

export function cleanProductUploaded (): { type: string } {
  return {
    type: actionTypes.CLEAN_UPLOAD_PRODUCT
  }
}
