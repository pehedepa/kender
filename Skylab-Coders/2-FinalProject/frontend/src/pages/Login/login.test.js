import React from 'react'
import LoginView from './index'
import { Alert } from 'react-native'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import * as actions from '../../redux/actions/actionCreators'
import { render, fireEvent } from '@testing-library/react-native'
import thunk from 'redux-thunk'

describe('Given a LoginView function', () => {
  const mockStore = configureStore([thunk])
  let navigation
  let store
  beforeEach(() => {
    jest.spyOn(actions, 'cleanLogin').mockReturnValueOnce({ type: '' })
    jest.spyOn(actions, 'login').mockReturnValueOnce({ type: '' })
    store = mockStore({
      statusLogin: 200
    })
    navigation = {
      replace: jest.fn(),
      navigate: jest.fn()
    }
  })
  describe('When we press Login Button with email and password with string values', () => {
    test('Then the render should match its snaptshot', () => {
      jest.spyOn(Alert, 'alert').mockReturnValueOnce({})
      store = mockStore({
        statusLogin: 401
      })

      const tree = render(<Provider store={store}><LoginView navigation={navigation}></LoginView></Provider>)
      const email = tree.getByPlaceholderText('Email')
      fireEvent.changeText(email, { text: 'email' })

      const password = tree.getByPlaceholderText('Password')
      fireEvent.changeText(password, { text: 'password' })

      const login = tree.getByTestId('Login')
      fireEvent.press(login)
      expect(tree).toMatchSnapshot()
    })
  })
  describe('When statusLogin is 401', () => {
    test('Then the render should match its snaptshot', () => {
      jest.spyOn(Alert, 'alert').mockReturnValueOnce({})
      store = mockStore({
        statusLogin: 401
      })

      const tree = render(<Provider store={store}><LoginView navigation={navigation}></LoginView></Provider>)
      expect(tree).toMatchSnapshot()
    })
  })
  describe('When Button Login is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const tree = render(<Provider store={store}><LoginView navigation={navigation}></LoginView></Provider>)
      fireEvent.press(tree.getByTestId('CreateAcc'))
      expect(navigation.navigate).toHaveBeenCalledWith('CreateAcc')
    })
  })
  describe('When email has no value', () => {
    test('Then an alert should pop up', () => {
      const mockState = React.useState
      jest.spyOn(React, 'useState')
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState(''))

      jest.spyOn(Alert, 'alert').mockReturnValueOnce({})
      store = mockStore({
        statusLogin: 401
      })

      const tree = render(<Provider store={store}><LoginView navigation={navigation}></LoginView></Provider>)
      fireEvent.press(tree.getByTestId('Login'))

      expect(Alert.alert).toHaveBeenCalled()
    })
  })
  describe('When email has value and password has no value', () => {
    test('Then an alert should pop up', () => {
      const mockState = React.useState
      jest
        .spyOn(React, 'useState')
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState('password'))

      jest.spyOn(Alert, 'alert').mockReturnValueOnce({})

      const tree = render(<Provider store={store}><LoginView navigation={navigation}></LoginView></Provider>)
      fireEvent.press(tree.getByTestId('Login'))

      expect(Alert.alert).toHaveBeenCalled()
    })
  })
})
