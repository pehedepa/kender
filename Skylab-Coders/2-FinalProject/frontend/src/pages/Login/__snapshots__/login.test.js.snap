// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginView function When statusLogin is 401 Then the render should match its snaptshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "resizeMode": "contain",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "padding": 25,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "80%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
      <View
        style={
          Object {
            "marginVertical": 10,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="email"
        maxLength={30}
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontFamily": "MontserratMedium",
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        maxLength={45}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontFamily": "MontserratMedium",
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "marginVertical": 10,
          }
        }
      />
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Login"
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              disabled={false}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FF988F",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginVertical": 10,
            }
          }
        />
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#005C85",
                "textAlign": "center",
              }
            }
            testID="CreateAcc"
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Given a LoginView function When we press Login Button with email and password with string values Then the render should match its snaptshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "resizeMode": "contain",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "padding": 25,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "80%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
      <View
        style={
          Object {
            "marginVertical": 10,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="email"
        maxLength={30}
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontFamily": "MontserratMedium",
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value={
          Object {
            "text": "email",
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        maxLength={45}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#e0e0e0",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontFamily": "MontserratMedium",
            "marginBottom": 15,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value={
          Object {
            "text": "password",
          }
        }
      />
      <View
        style={
          Object {
            "marginVertical": 10,
          }
        }
      />
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="Login"
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              disabled={false}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#FF988F",
                  },
                ]
              }
            >
              Login
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginVertical": 10,
            }
          }
        />
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#005C85",
                "textAlign": "center",
              }
            }
            testID="CreateAcc"
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
